libs/community/langchain_community/chat_message_histories/sql.py

```diff
@@ -80,36 +92,85 @@ def get_sql_model_class(self) -> Any:
         return self.model_class
 
 
+DBConnection = Union[AsyncEngine, Engine, str]
+
+
 class SQLChatMessageHistory(BaseChatMessageHistory):
     """Chat message history stored in an SQL database."""
 
+    @property
+    def Session(self) -> Union[scoped_session, async_sessionmaker]:
+        logger.warning("Session is deprecated, use session_maker instead")
```

Comment:
Could you use @deprecated decorator on top of the property if you want to deprecate this. It's in `langchain_core._api`. 

It has logic for:
1) Not raising warnings for internal calls to this code that the user has no control over (i.e., from within langchain code)
2) only showing the warning once
3) It uses a warning rather than logger.warning because in this case moving away from deprecated code is under control over the user

4) We can use it to auto-generate list of deprecation tables when publishing release notes (e.g., https://python.langchain.com/v0.2/docs/versions/v0_2/deprecations/)

Reply:
done